changelog.d/21201_fields_iter.fix.md

```diff
@@ -0,0 +1 @@
+The fields iterator now quotes all invalid path segments and he reduce transform can correctly parse path containing special characters.
```

Comment:
```suggestion
The `reduce` transform no longer panics when reducing fields that contain special characters.
```

Nit: I think this will be a bit more understandable to Vector users. The bits about the iterator are only relevant to developers, who will have the commit logs.